src/wp-includes/user.php

```diff
@@ -204,11 +204,11 @@ function wp_authenticate_email_password( $user, $email, $password ) {
 		return $user;
 	}
 
-	if ( empty( $email ) || empty( $password ) ) {
-		if ( is_wp_error( $user ) ) {
-			return $user;
-		}
+	if ( is_wp_error( $user ) && ! filter_var( $email, FILTER_VALIDATE_EMAIL ) ) {
```

Comment:
We don't use `filter_var` in Core, filter is an optional extension. If this is needed, it should be `is_email` instead.

Reply:
```suggestion
	if ( is_wp_error( $user ) && ! is_email( $email ) ) {
```

Reply:
The fact that we need this check in the first place makes me a little nervous. AFAICT we are including this because otherwise when `wp_authenticate_username_password` returns an error we would short-circuit the email based authentication. 

What happens if another auth mechanism returns an error when someone is logging in with an email address? Won't we end up swallowing their error still?

Reply:
I'm not convinced yet that we do need this check. I've removed it to see what happens. Will test without it. If it's needed, then like you @TimothyBJacobs, something is wrong that needs further investigation.

Reply:
@TimothyBJacobs @costdev I just confirmed via manual testing that it works without adding an additional is_email check. If the tests fail, I'll look and adjust them.

Reply:
Hmm I see the problem. The change of pulling the error before the empty checks means users can't log in via email. Not good.